### ISSUE TITLE
Max-Length-Validierung goes wrong for decoded entities

### ISSUE THREAD

**Issue by Defcon0** [contao/core#8188]
Hello folks,

when having let's say a textarea defining a eval->maxlength-dca property of 400, validation goes wrong if the text contains encodable characters like "(" and ")" which in fact are encoded to &#40; and &#41;. This encoding increases the string length the Widget::validator method uses to validate the max length.

I added an additional html_entity_decode() call to fix this:

if ($this->minlength && $varInput != '' && utf8_strlen(html_entity_decode($varInput)) < $this->minlength)
		{
			$this->addError(sprintf($GLOBALS['TL_LANG']['ERR']['minlength'], $this->strLabel, $this->minlength));
		}

		if ($this->maxlength && $varInput != '' && utf8_strlen(html_entity_decode($varInput)) > $this->maxlength)
		{
			$this->addError(sprintf($GLOBALS['TL_LANG']['ERR']['maxlength'], $this->strLabel, $this->maxlength));
		}

**Comment by leofeyer** [comment]
I can confirm this. It is a bit awkward that you enter something shorter than the maximum length and still get the error message; still we cannot apply the fix above, because the encoded string will be written to the database and must therefore not exceed the maximum length.

@contao/developers /cc

**Comment by Toflar** [comment]
Yeah it's strange but it is what it is. There's no solution to that except for adding this hint to the error language string (which in 99.9% of all cases will just confuse people).

**Comment by discordier** [comment]
Side note: what is about utf-8 here? Is max length in ASCII or utf-8? Could that also raise the issue?

**Comment by leofeyer** [comment]
Since we are using `utf8_strlen()`, the answer should be obvious :)

**Comment by discordier** [comment]
you are right, missed that while reading the issue on mobile.
IMO this renders the check useless then, as the length is most likely to be used to determine if the value fits into a database field, am I wrong on this one?

Anyway, the proposed addition of `html_entity_decode` will potentially also decode `&lt;foooo bar&gt;` which will then get killed by the strip tags and therefore make a really long string very short.
Therefore this is not an option.

**Comment by leofeyer** [comment]
As discussed in Mumble on April 21st, this problem can only be resolved in Contao 5, when we switch from input decoding to output decoding.
